### PR TITLE
[py] Update timeouts.py docstrings

### PR DESCRIPTION
**Pull request by palmermbandy**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description
Updated the docstrings in `py/selenium/webdriver/common/timeouts.py` to fix some typos and add descriptions for each attribute of the `Timeouts` class (I used the corresponding descriptions from `rb/lib/selenium/webdriver/common/timeouts.rb`).

### Motivation and Context
Ensure accurate and informative docstrings.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: py/selenium/webdriver/common/timeouts.py [hunk 1]
```diff
@@ -60,40 +60,42 @@ def __set__(self, obj, value) -> None:
...
+        - `self.implicit_wait` = `value`
 
     Parameters
     ----------
     `value`: `float`
     """
 
     page_load = _TimeoutsDescriptor("_page_load")
-    """Sets and Gets the value of page load wait:
+    """Sets and Gets the value of page_load:
```
Comment (titusfortner) [inline py/selenium/webdriver/common/timeouts.py]: I think these should all be English not code (e.g., "page load timeout")

**Review by titusfortner — CHANGES_REQUESTED** [review]
I love adding the w3c link to the docs, thanks for renaming the implicit_timeout since it was inconsistent. I think best practice for docstrings is to be regular english? I'm not an expert, but seems more descriptive to say "script execution timeout" and "page load timeout on navigation" etc etc

File: py/selenium/webdriver/common/timeouts.py (hunk 1 above)
Comment (AutomatedTester) [reply py/selenium/webdriver/common/timeouts.py]: Agreed. This should be for humans to read. They won't know what `page_load` means

Comment (palmermbandy) [reply py/selenium/webdriver/common/timeouts.py]: 👍 I'll update these.

**Review by titusfortner — APPROVED** [review]

**Comment by isaulv** [comment]
This should be good to go.

**Comment by titusfortner** [comment]
@palmermbandy thank you!

**Pull request merged**